Sources/Async/Streams/ByteSerializerStream.swift

```diff
@@ -0,0 +1,129 @@
+public protocol ByteSerializerStream: Async.Stream, ConnectionContext where Output == UnsafeBufferPointer<UInt8> {
+    var state: ByteSerializerStreamState<Self> { get }
+    
+    func serialize(_ input: Input) -> UnsafeBufferPointer<UInt8>
```

Comment:
how exactly do you implement this function? are you required to manipulate the `state` variable manually or is it purely just serializing the message? what happens if the message is bigger than the buffer?

Reply:
The message is never bigger than the buffer. The buffer must be available asynchronously and is provided by the serializer. I'll document that

Reply:
Oh, I get it now. You mean multiple writes